statics/workflows/ping-mesh.yaml

```diff
@@ -0,0 +1,103 @@
+---
+UUID: "1d81b961-5d62-49b7-42f3-c6afc616ff98"
+name: "PingMesh(ICMP/TCP/UDP)"
+description: "Check Connectivity from Multiple Source to Single Destination"
+parameters:
+  - name: source
+    description: Source node
+    type: node
+  - name: destination
+    description: Destination node
+    type: node
+  - name: protocol
+    description: Protocol
+    type:  choice
+    default: ICMPv4
+    values:
+      - description: "Protocol : ICMPv4/Echo request"
+        value: ICMPv4
+      - description: "Protocol : TCP/IPv4"
+        value: TCP
+      - description: "Protocol : UDP/IPv4"
+        value: UDP	
+source: |
+    function PingMesh(from, to, protocol) {
+      var dict = {},
+            From = {},
+            sources = [],
+            captureuuids = [],
+            i = 1,
+            randvalue = Math.floor(Math.random() * 4096);
+      var capture = new Capture();
+      var packetInjection = new PacketInjection();
+      sources.push(from);
+      sources.forEach(function(source){
+        capture.GremlinQuery = "G.V().Has('TID', '" + source + "').ShortestPathTo(Metadata('TID', '" + to + "'))";
+        packetInjection.Src = "G.V().Has('TID', '" + source + "')"
+        packetInjection.Dst = "G.V().Has('TID', '" + to + "')"
+        packetInjection.Count = 5
+        return client.captures.create(capture).then(function (c) {
+            capture = c
+            captureuuids.push(capture.UUID);
+            return sleep(2000)
+        }).then(function () {
+            return client.G.V().Has("TID", source)
+        }).then(function (nodes) {
+            if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+                packetInjection.SrcIP = nodes[0].Metadata.Neutron.IPV4[0]
+            }
+            if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+                packetInjection.SrcMAC = nodes[0].Metadata.ExtID["attached-mac"]
+            }
+            if (protocol == "ICMPv4") {
+                packetInjection.Type = "icmp4"
+                packetInjection.ICMPID = randvalue + i;
+                From[source] = packetInjection.ICMPID;
+            }
+            if (protocol == "TCP") {
+                packetInjection.Type = "tcp4"
+                packetInjection.SrcPort = 16384+randvalue + i;
```

Comment:
save the src and dst port in the source map, to correlate/search the flow later

srcPort = 1+(Math.random() * (65534-1024))
dstPort = 1+(Math.random() * (65534-1024))